src/benchmarking.rs

```diff
@@ -4,17 +4,359 @@ use super::*;
+    }: _(RawOrigin::Signed(caller.clone()), PERMISSION_ID.clone(), ROLE_ID.clone())
+    verify {
+        assert_last_event::<T>(Event::<T>::PermissionUnassignedToRole(
+            caller.clone(),
+            PERMISSION_ID.clone(),
+            ROLE_ID.clone(),
+        ).into());
+    }
+
+    fetch_role_permissions {
+        let caller : T::AccountId = account(CALLER_ACCOUNT_STR, 0, 0);
+        RBAC::<T>::add_role(RawOrigin::Signed(caller.clone()).into(), ROLE_ID.clone(), ADMIN_STR.to_vec())?;
+        RBAC::<T>::add_permission(
+            RawOrigin::Signed(caller.clone()).into(), PERMISSION_ID.clone(), ADMIN_STR.to_vec())?;
+        RBAC::<T>::assign_permission_to_role(
+            RawOrigin::Signed(caller.clone()).into(), PERMISSION_ID.clone(), ROLE_ID.clone())?;
+    }: _(RawOrigin::Signed(caller.clone()), caller.clone(), ROLE_ID.clone())
+    verify {
+    }
+
+    add_group {
+        let caller : T::AccountId = account(CALLER_ACCOUNT_STR, 0, 0);
+    }: _(RawOrigin::Signed(caller.clone()), GROUP_ID.clone(), GROUP_STR.to_vec())
+    verify {
+        assert_last_event::<T>(Event::<T>::GroupAdded(
+            caller.clone(),
+            GROUP_ID.clone(),
+            GROUP_STR.to_vec(),
+        ).into());
+    }
+
+    update_group {
+        let caller : T::AccountId = account(CALLER_ACCOUNT_STR, 0, 0);
+        RBAC::<T>::add_group(
+            RawOrigin::Signed(caller.clone()).into(), GROUP_ID.clone(), GROUP_STR.to_vec())?;
+    }: _(RawOrigin::Signed(caller.clone()), GROUP_ID.clone(), GROUP_STR.to_vec())
+    verify {
+        assert_last_event::<T>(Event::<T>::GroupUpdated(
+            caller.clone(),
+            GROUP_ID.clone(),
+            GROUP_STR.to_vec(),
+        ).into());
+    }
+
+    disable_group {
+        let caller : T::AccountId = account(CALLER_ACCOUNT_STR, 0, 0);
+        RBAC::<T>::add_group(
+            RawOrigin::Signed(caller.clone()).into(), GROUP_ID.clone(), GROUP_STR.to_vec())?;
+    }: _(RawOrigin::Signed(caller.clone()), GROUP_ID.clone())
+    verify {
+        assert_last_event::<T>(Event::<T>::GroupDisabled(
+            caller.clone(),
+            GROUP_ID.clone(),
+        ).into());
+    }
+
+    fetch_group {
+        let caller : T::AccountId = account(CALLER_ACCOUNT_STR, 0, 0);
+        RBAC::<T>::add_group(
+            RawOrigin::Signed(caller.clone()).into(), GROUP_ID.clone(), GROUP_STR.to_vec())?;
+    }: _(RawOrigin::Signed(caller.clone()), caller.clone(), GROUP_ID.clone())
     verify {
-        assert_eq!(RoleStore::<T>::get(), Some(s));
     }
+
+    fetch_groups {
+        let caller : T::AccountId = account(CALLER_ACCOUNT_STR, 0, 0);
+        RBAC::<T>::add_group(
+            RawOrigin::Signed(caller.clone()).into(), GROUP_ID.clone(), GROUP_STR.to_vec())?;
+        RBAC::<T>::add_group(
+            RawOrigin::Signed(caller.clone()).into(), GROUP_ID2.clone(), GROUP_STR.to_vec())?;
+    }: _(RawOrigin::Signed(caller.clone()), caller.clone())
+    verify {
```

Comment:
Why are we not verifying all fetch extrinsics benchmark?

Reply:
Let me add them.

Reply:
In this Marco, it'll check the return value automatically. If the "EntityDoesNotExist" occurs, the benchmark fails. Therefore, we can remove the verify block in the fetch-related functions.